File: 1514-minimum-value-to-get-positive-step-by-step-sum/minimum-value-to-get-positive-step-by-step-sum.py
class Solution:
    def minStartValue(self, nums: List[int]) -> int:
        min_prefix = float("inf")
        total = 0
        for i in range(len(nums)):
            total += nums[i]
            min_prefix = min(min_prefix, total)
        
        start_value = 1
        while start_value + min_prefix < 1:
            start_value += 1
        return start_value

# [-3,2,-3,4,2]
# prefix = [-3, -1, -4, 0, 2]

# [1,-2,-3]
# prefix = [1, -1, -4]

[2,5,10,5,4]

File: 1514-minimum-value-to-get-positive-step-by-step-sum/minimum-value-to-get-positive-step-by-step-sum.cpp
class Solution {
public:
    int minStartValue(vector<int>& nums) {
        int min_val = 1;
        int total = 0;
        for (int num : nums) {
            total += num;
            min_val = min(min_val, total);
        }
        
        return (min_val >= 1) ? min_val : abs(min_val) + 1;
    }
};

